### PR TITLE
academy/blocks/topic teaser block WITH MIGRATION

### PR DESCRIPTION
**Pull request by khamui**
~~There is no need to make the categories/topic editable (as PM stated).
But not sure if this simple way is what we want. Should we somehow get the categories/topics that are defined in `AcademyPage`?~~

~~The Images have to be added somewhere too. For now I just simply leave it to the Editors, to upload those.~~

### CODE REVIEW

File: apps/academy/models.py [hunk 1]
```diff
@@ -15,7 +15,12 @@
 from wagtail.models import Page
 from wcag_contrast_ratio import contrast
 
+<<<<<<< HEAD
```
Comment (fuzzylogic2000) [inline apps/academy/models.py:18]: This shouldn't be here.

**Review by fuzzylogic2000 — COMMENTED** [review]
I have difficulties reviewing this as I think there is stuff in from other old commits? (And I normally go through the commits one by one to understand what was done.)
Could you please put all the changes in one commit here as I think that might be easiest to understand in the end?

File: apps/academy/models.py (hunk 1 above)
Comment (fuzzylogic2000) [reply apps/academy/models.py:18]: And the other bits in the lines after from the failed rebase or merge as well.

File: apps/academy/blocks.py [hunk 2]
```diff
@@ -1,7 +1,11 @@
+from django.template.defaulttags import register
```
Comment (khamui) [inline apps/academy/blocks.py:1]: three parts just to get the right name of the category.

pt1

File: apps/academy/blocks.py [hunk 3]
```diff
@@ -56,3 +60,37 @@ class Meta:
...
+    title = CharBlock(required=True, length=74)
+    topics = ListBlock(TopicBlock(), min_num=3, max_num=3)
+    topic_url = PageChooserBlock(
+        required=True,
+        page_type='academy.AcademyIndexPage'
+    )
+
+    def get_context(self, value, parent_context=None):
+        context = super().get_context(value, parent_context=parent_context)
+        context['topics_dict'] = dict(TOPIC_CHOICES)
```
Comment (khamui) [inline apps/academy/blocks.py:85]: pt2 - category name

File: apps/academy/blocks.py (hunk 2 above)
Comment (khamui) [reply apps/academy/blocks.py:1]: that is not elegant, but i couldn't find another way

**Comment by khamui** [comment]
Note: migrations removed again. To be added :)

**Comment by fuzzylogic2000** [comment]
@Rineee Maybe you can have a look? Then I will go on with my stuff instead of reviewing this. (But wait with migrations and merge for @phillimorland PR, which is nearly done.)

File: apps/academy/blocks.py [hunk 4]
```diff
@@ -56,3 +60,37 @@ class Meta:
...
         label = 'Single teaser block'
+
+
+class TopicBlock(StructBlock):
+    topic_category = ChoiceBlock(
+        choices=TOPIC_CHOICES,
+        required=True,
+        help_text='Select a topic'
+    )
+    topic_text = CharBlock(required=True, length=110)
```
Comment (fuzzylogic2000) [inline apps/academy/blocks.py:71]: I copied the length from core/blocks and then we put it everywhere. Should be max_length!

**Review by Rineee — APPROVED** [review]
Nice!

**Pull request merged**